### DISCUSSION TITLE
User Actions

### DISCUSSION THREAD

**Discussion by midsplit** [tidal-music/discussions#15] [Ideas & Feature requests]
It would be interesting to be able to collect auth and refresh tokens from end users with their approval to perform actions on their behalf on Tidal such as: Saving albums/tracks to their library, following artists, managing playlists, and possibly more interesting use-cases the community can come up with.

For example, on Releese we can enable artists to create dedicated Pre-save pages for Spotify, Apple Music, and Deezer, on which an end user of those platforms can login with their account, granting us a token for future use, which we use to save a specific release to their library. It would be wonderful for our community of creators to be able to add Tidal to their list of Pre-save providers.

We are aware that some other platforms (Feature.fm, LinkFire) can achieve this action through other means but it would be interesting for this to be available in the API.

**Comment by stefanmoro** [comment]
Hi @midsplit, and welcome to the forum.

This is functionality we have just started investigating. As you already touched upon in your post, making sure that the end user stays in control of their data is very important and something we are always focusing on.

This functionality could hopefully help unlock various new features for apps built on our API. And so it is also important that we work closely together with other stakeholders (e.g. labels, artists and content owners) to make sure the solutions we build are beneficial for all involved parties. 

We'll post any updates here once we have them.

Thanks

**Comment by ackmanx** [comment]
This would be really nice to have! I currently use the Spotify API to get my own followed artists so I can query for their list of albums and it's currently the single one thing that keeps me on that platform

**Reply by stefanmoro** [reply]
This is possible to do. Simply sign up on developer.tidal.com, create a client and query the [artists collection](https://tidal-music.github.io/tidal-api-reference/#/userCollections/get_userCollections__id__relationships_artists) endpoint to get your artists. You can then follow the `albums` relationship to get the albums.

**Reply by ackmanx** [reply]
Oh, fantastic! I guess I didn't understand what the relationship endpoint did when I was reading the API. Thanks much, I just tried and it worked great. I now have no reasons left to stay on Spotify 😆
